packages/amplify_auth_cognito/test/amplify_auth_cognito_authExceptions_test.dart

```diff
@@ -33,14 +33,14 @@ void main() {
 
   tearDown(() {});
 
-
   test('castAndThrowPlatformException returns an AuthException', () async {
-    PlatformException exception = PlatformException(code: "AuthException", details: details);
-    AmplifyException amplifyException;
+    PlatformException exception =
+        PlatformException(code: "AuthException", details: details);
+    late AmplifyException amplifyException;
     try {
-      await castAndThrowPlatformException(exception);
+      castAndThrowPlatformException(exception);
     } catch (e) {
-      amplifyException = e;
+      amplifyException = e as AmplifyException;
```

Comment:
Interesting point there.  It's necessary for amplifyException to be of type AmplifyException because that class has the fields message, recoverySuggestion, and underlyingException which the Exception class does not have.  So at some point we should have a variable of type AmplifyException to avoid needing to cast to AmplifyException each time we want to access those fields. 

Great side note as well.  We have calls to isInstanceOf throughout the code and I don't know if it's within scope of the Auth PR to make all those changes.  Perhaps let's create a separate PR into the Null Safety Master branch to track changes for that.  What do you think @cshfang

Reply:
Yep I think that a separate PR to address that makes the most sense.

Reply:
+1 this seems a good way to fix the unintuitive-ness that originally raised my eyebrow here. What I am uncertain of is if it should be addressed in this PR or if we should have another round of PRs to specifically address this.

Reply:
Sounds good, updated all Auth unit tests